src/main/java/net/neoforged/neoforge/client/model/data/ModelData.java

```diff
@@ -80,7 +80,7 @@ public <T> Builder with(ModelProperty<T> property, T value) {
 
         @Contract("-> new")
         public ModelData build() {
-            return new ModelData(Collections.unmodifiableMap(properties));
+            return new ModelData(properties.isEmpty() ? Collections.emptyMap() : Collections.unmodifiableMap(properties));
```

Comment:
There should be a comment explaining why this is done, it's not obvious that `Collections.emptyMap()` is faster because it doesn't hash. Also, we could use `Map.of()`. :P